lib/cpp/scalgoproto.hh

```diff
@@ -827,7 +907,7 @@ public:
 	}
 
 	~Writer() {
-		if (data) free(data);
+		setCapacity(0);
```

Comment:
I don't see any where we truncate the final file, to have a size that is approximatly size. It would probably be a good idea to do that here, in case the file is copied in an non hole aware way.

Reply:
Thanks - I've extended the API of the WriterBacking so that it is truncated when finalize() is called. Please take a look.